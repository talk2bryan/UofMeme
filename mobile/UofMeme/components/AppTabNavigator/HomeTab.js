import React, { Component } from "react";
import { View, Text, StyleSheet, Button } from "react-native";
import { Container, Content, Icon } from "native-base";

import CardComponent from "../CardComponent";

class HomeTab extends Component {
  render() {
    return (
      <Container>
        <Content>
          <CardComponent imageSource="1" likes="9001" />
          <CardComponent imageSource="2" likes="623" />
          <CardComponent imageSource="3" likes="8999" />
        </Content>  
      </Container>
    );
  }
}

export default HomeTab;
