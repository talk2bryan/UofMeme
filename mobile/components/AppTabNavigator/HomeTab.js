import React, { Component } from "react";
import { View, Text, StyleSheet } from "react-native";

import { Container, Content, Icon } from "native-base";

import CardComponent from "../CardComponent";

class HomeTab extends Component {
  static navigationOptions = {
    tabBarIcon: ({ tintColor }) => (
      <Icon name="ios-home" style={{ color: tintColor }} />
    )
  };

  render() {
    return (
      <Container style={styles.container}>
        <Content>
          <CardComponent imageSource="1" likes="9001" />
          <CardComponent imageSource="2" likes="623" />
          <CardComponent imageSource="3" likes="8999" />
        </Content>
      </Container>
    );
  }
}

//make this component available to the app
export default HomeTab;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white"
  }
});
