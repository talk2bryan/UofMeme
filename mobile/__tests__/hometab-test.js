import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import "isomorphic-fetch";

import HomeTab from '../components/AppTabNavigator/HomeTab';

it('renders correctly', () => {
  const tree = renderer.create(
    <HomeTab />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});