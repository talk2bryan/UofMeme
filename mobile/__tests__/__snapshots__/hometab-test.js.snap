// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingTop": 20,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
